lib/ZnapZend.pm

```diff
@@ -99,7 +99,8 @@ has zLog => sub {
     return $log;
 };
 
-my $killThemAll  = sub {
+### private methods ###
+our $killThemAll  = sub {
```

Comment:
The point of using `my` is to 'hide' the method from external access. By using `our` you make the methods publicly accessible ...  so you may want to do these changes just in the code that modifies the source at runtime before executing the tests.

Reply:
Huh, is there some such feature? :-)

Could you please close this PR and do it the right way then, I don't know how to approach it this way properly and am getting bogged on other PRs and work that I do have ideas about :)

Reply:
in the test code there is a function which modifies the perl code prior to runing the tests ... these modifications are the root of the warnings ... that is why they only occur while runnig the tests ... so the solution you found would be fine but it has to be at runtime in this code ... https://github.com/jimklimov/znapzend/blob/unshared-vars/t/znapzend.t#L20

Reply:
Oh wow :) I thought it was due to the `use strict; use warnings;` which are not used in production code under `bin/` but are enforced by import into test scripts.

Reply:
Hopefully got it done :)